src/storage/secondary/transaction.rs

```diff
@@ -70,15 +67,15 @@ impl SecondaryTransaction {
         update: bool,
     ) -> StorageResult<Self> {
         // pin a snapshot at version manager
-        let (epoch, snapshot) = table.version.pin();
+        let version = table.version.pin();
```

Comment:
Should store version inside txn struct, otherwise it will get immediately unpinned once txn is created.